avm/res/web/site/tests/e2e/webApp.defaults/main.test.bicep

```diff
@@ -57,9 +57,15 @@ module testDeployment '../../../main.bicep' = [
       siteConfig: {
         healthCheckPath: '/healthz'
         alwaysOn: true
+        ftpsState: 'FtpsOnly'
```

Comment:
The values you added - are they all now mandatory with the latest update, or did you set them because PSRule was complaining about them?

I strongly suspect the latter. There is a spec in AVM somewhere for it, but to avoid looking it up, let me elaborate here
- The `waf-aligned` tests should contain every parameter we'd need to deploy the module in a waf-compliant way (and ideally no more)
- The `defaults` test in contrast should deploy the module with only the required parameters so that the defaults are used. These defaults should hence be WAF-compliant.
Now here's the catch: That would mean the latter may fail some PSRules. To address those there are 2 options:
- If it's a simple value that can have a default (e.g., `localAuth: false`) we should set it in the template file
- If not - then we should add it to the suppresion rules for the default tests you can find [here](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/min-suppress.Rule.yaml). We don't exactly advertise it, as it should be a 'last resort'

Now here's the double-catch. When we applied this strategy we soon found a handful of cases where one may wonder if some of the rules should be applied in general. I think you found one of those which says 'resource xzy should have a managed identity' or similar. The thought of the rule is good - if 2 services should communicate they should use a managed identity. But not every service you deploy communicates outbound, so what's the point of enforcing a managed identity? These would either also go into the suppresion rules as you can see for example here
https://github.com/Azure/bicep-registry-modules/blob/b5d76b9c9931ba130db315643edcfc064e855bd2/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/min-suppress.Rule.yaml#L17
and or should be called out in the PSRule repository as a rule that should be reconsidered. For example [here](https://github.com/Azure/Azure-Verified-Modules/issues/349#issuecomment-2061504618). 
We're collaborating closely with @BernieWhite to provide feedback and help with the WAF-compliance tests, and naturally, also your feedback is well received.

Reply:
Same feedback for the `webAppLinux.defaults` test

Reply:
@AlexanderSehr for App Service I feel this should be required...there aren't many scenarios where a web service isn't contacting some data source.

I agree the defaults should be updated...would consider this case the some of these are just good practice...minTLS, HTTP2, SFTP. The question/concern/feedback before making a change to this is who/whom/how should determine this?

Reply:
That would be the module owner(s). In this case Sir @tsc-buddy to the rescue.

Processwise, module owners (unless orphaned) are automatically added to PRs to their modules to review. The only cases where the CI maintainers may bypasss this are batch updates (if we find for example an issue with an implementation existing in many modules that we fix at once).

Reply:
Also beware a sizable amount of conflicts for this PR (likely due to the recently merged fixed for the WAF-reliablity tests).

My advice
- Disable actions in your fork's settings
- Merge upstream & resolve conflicts
- Enable actions again

As your PR comes directly from main, you may need to do the conflict resolution via your local termal + VSCode though. I doubt GitHub will be keen on merging upstream in the current setup.

Reply:
Conflicts are resolved now